store/store_action.go

```diff
@@ -309,7 +314,7 @@ type AssertionResult struct {
 	StreamURLs []string
 }
 
-func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, actions []*SnapAction, assertQuery AssertionQuery, toResolve map[asserts.Grouping][]*asserts.AtRevision, toResolveSeq map[asserts.Grouping][]*asserts.AtSequence, user *auth.UserState, opts *RefreshOptions) ([]SnapActionResult, []AssertionResult, error) {
+func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, actions []*SnapAction, assertQuery AssertionQuery, toResolve map[asserts.Grouping][]*asserts.AtRevision, toResolveSeq map[asserts.Grouping][]*asserts.AtSequence, user *auth.UserState, opts *RefreshOptions, storeVer *int) ([]SnapActionResult, []AssertionResult, error) {
```

Comment:
Taking a pointer makes the zero value more obvious but I think an `int` would've been fine. We could ignore if it was zero (since no proxy will likely return it) or negative and take it into account otherwise. Just a opinion, maybe @pedronis disagrees

Reply:
I am thinking that could allow malicious servers (mitm) to use this header to force snapd into an infinite recursion, and eventually crashing. for example always setting it to 0, or not setting the header at all. I actually noticed this when some tests crashed due not setting the snap version header as it was not relevant to their test case.
That was my reasoning when opting for passing a pointer.

Reply:
IMO we should have an explicit limit to retries. Right now, retries are limited because `strconv.Atoi` returns a zero on error (which we would get if the header was empty), which we take a reference of and then check against nil. That's not very clear IMO and my (personal) preference would be to be very explicit about it and check that the header carries a version that's worth retrying with. But we can wait and see if other reviewers feel differently. 
Also if we consider malicious server, we might have many other problems so I'm not sure that's something we usually include in our fault model.

Reply:
> my (personal) preference would be to be very explicit about it and check that the header carries a version that's worth retrying with.

Aha, I like this approach, thank you for explaining.